nncf/experimental/post_training/api/engine.py

```diff
@@ -11,38 +11,128 @@
  limitations under the License.
 """
 
-from abc import ABC
 from abc import abstractmethod
 
 from typing import Dict
 from typing import TypeVar
+from typing import Callable
+from nncf.experimental.post_training.api.sampler import Sampler
 
 ModelType = TypeVar('ModelType')
 TensorType = TypeVar('TensorType')
+MetricType = TypeVar('MetricType')
 
 
-class Engine(ABC):
+class Engine:
```

Comment:
It's still ABC, because it doesn't make sense to create an instance of `Engine`, cause it doesn't have `infer()`

Reply:
Fixed to inherit from ABC